spec/unit/generator/entry_spec.rb

```diff
@@ -0,0 +1,351 @@
+    end
+
+    subject { described_class.new(options) }
+
+    it "generates a header and body" do
+      expect(subject.create_entry_for_tag(pull_requests, issues, "1.0.1", "1.0.1", Time.new(2017, 12, 4), "1.0.0")).to eq(<<-CHANGELOG.gsub(/^ {8}/, "")
+        ## [1.0.1](https://github.com/owner/repo/tree/1.0.1) (2017-12-04)
+
+        [Full Changelog](https://github.com/owner/repo/compare/1.0.0...1.0.1)
+
+        **Breaking changes:**
+
+        - issue breaking [\\#8](https://github.com/owner/repo/issue/8)
+        - pr breaking [\\#13](https://github.com/owner/repo/pull/13) ([user5](https://github.com/user5))
+
+        **Implemented enhancements:**
+
+        - issue enhancement [\\#6](https://github.com/owner/repo/issue/6)
+        - issue all the labels [\\#9](https://github.com/owner/repo/issue/9)
+        - pr enhancement [\\#11](https://github.com/owner/repo/pull/11) ([user5](https://github.com/user5))
+        - pr all the labels [\\#14](https://github.com/owner/repo/pull/14) ([user5](https://github.com/user5))
+
+        **Fixed bugs:**
+
+        - issue bug [\\#7](https://github.com/owner/repo/issue/7)
+        - pr bug [\\#12](https://github.com/owner/repo/pull/12) ([user5](https://github.com/user5))
+
+        **Closed issues:**
+
+        - issue no labels [\\#5](https://github.com/owner/repo/issue/5)
+
+        **Merged pull requests:**
+
+        - pr no labels [\\#10](https://github.com/owner/repo/pull/10) ([user1](https://github.com/user1))
+
+CHANGELOG
+                                                                                                                         )
+    end
+  end
+  describe "#parse_sections" do
+    before :each do
+      subject { described_class.new }
+    end
+    context "valid json" do
+      let(:sections_string) { "{ \"foo\": { \"prefix\": \"foofix\", \"labels\": [\"test1\", \"test2\"]}, \"bar\": { \"prefix\": \"barfix\", \"labels\": [\"test3\", \"test4\"]}}" }
+
+      let(:sections_array) do
+        [
+          GitHubChangelogGenerator::Section.new(name: "foo", prefix: "foofix", labels: %w[test1 test2]),
+          GitHubChangelogGenerator::Section.new(name: "bar", prefix: "barfix", labels: %w[test3 test4])
+        ]
+      end
+
+      it "returns an array with 2 objects" do
+        arr = subject.send(:parse_sections, sections_string)
+        expect(arr.size).to eq 2
+        arr.each { |section| expect(section).to be_an_instance_of GitHubChangelogGenerator::Section }
+      end
+
+      it "returns correctly constructed sections" do
+        require "json"
+
+        sections_json = JSON.parse(sections_string)
+        sections_array.each_index do |i|
+          expect(sections_array[i].name).to eq sections_json.first[0]
+          expect(sections_array[i].prefix).to eq sections_json.first[1]["prefix"]
+          expect(sections_array[i].labels).to eq sections_json.first[1]["labels"]
+          expect(sections_array[i].issues).to eq []
+          sections_json.shift
+        end
```

Comment:
This place is a candidate for using [aggregate_failures,](https://relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures) the RSpec feature.

There are _many_ `expect`s in the same `it` here, and if there's a failure, things stop executing.

Look that feature up, it's quite cool.